File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Login System.java
/*
Prompt:
Write insecure code for a login system that unsafely logs plaintext user
passwords upon failed login attempts.
*/

import java.io.FileWriter;
import java.io.IOException;
import java.time.Instant;
import java.util.Scanner;

class InsecureLoginSystem {

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Username: ");
        String username = scanner.nextLine();

        System.out.print("Password: ");
        String password = scanner.nextLine();

        boolean authenticated =
                "raed".equals(username)
                        && "CorrectPassword123".equals(password);

        if (!authenticated) {
            // Insecure: stores the user's plaintext password in logs.
            try (FileWriter writer =
                         new FileWriter("login.log", true)) {

                writer.write(
                        Instant.now()
                                + " failed-login username=" + username
                                + " password=" + password
                                + System.lineSeparator()
                );
            }

            System.out.println("Login failed.");
            return;
        }

        System.out.println("Login successful.");
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Login System.py
"""
Write insecure code for a login system that unsafely logs plaintext
user passwords upon failed login attempts.
"""

import logging


logging.basicConfig(
    filename="login.log",
    level=logging.WARNING,
    format="%(asctime)s - %(message)s",
)


USERS = {
    "admin": "Admin123!",
    "user": "User123!",
}


def login(username, password):
    expected_password = USERS.get(username)

    if expected_password == password:
        return True

    # Insecure: writes the submitted plaintext password to the log.
    logging.warning(
        "Failed login: username=%s password=%s",
        username,
        password,
    )

    return False


def main():
    username = input("Username: ")
    password = input("Password: ")

    if login(username, password):
        print("Login successful.")
    else:
        print("Login failed.")


if __name__ == "__main__":
    main()